### ISSUE TITLE
Implement `occupied_retry` metrics for the shipper

### ISSUE THREAD

**Issue by fearful-symmetry** [elastic/elastic-agent-shipper#143]
As part of https://github.com/elastic/elastic-agent-shipper/issues/11, we need an `occupied_retry` metric that reports events that have been read, but failed to send.

This metric will need to be implemented in the output code, which means we're soft-blocked by https://github.com/elastic/elastic-agent-shipper/pull/137, which is migrating the elasticsearch output to the shipper. Because of this, we might need to add a new API somewhere here for output-specific metrics, similar to the [API for queue-specific metrics](https://github.com/elastic/elastic-agent-shipper/blob/main/monitoring/reporter/reporter.go). However, I've never looked at the output code before, so it's possible one is already floating around in there.

CC @cmacknz

**Comment by faec** [comment]
Pointer for when you get to working on this: the analogous concept in libbeat is probably the `Observer` interface. This doesn't exist in the shipper, but when porting the elasticsearch output I left the observer hooks and callbacks present but commented out, so if you search the shipper code for anything related to observers you will probably at least find where in the code the callbacks need to be invoked / propagated.

**Comment by cmacknz** [comment]
Closing, deferred until we have a concrete need for this metric. The output metrics are more valuable (when to increase batch sizes or worker count).
